components/IndoorDirections/Dijkstra/DijkstraAlgorithm.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Implementation of Dijkstra's shortest path algorithm
+ * @param {*} graph | graph of nodes (ClassGraph.js)
+ * @param {*} start | starting node
+ * @param {*} end | ending node
+ */
+export const dijkstra = (graph, start, end) => {
+    // track the lowest cost to reach each node
+    let costs = {};
+    costs[end] = "Infinity";
+    costs = Object.assign(costs, graph[start]);
+
+    // track paths
+    const parents = {end: null};
+    for (let child in graph[start]) {
+        parents[child] = start;
+    }
+
+    const processed = [];
+    let node = lowestCostNode(costs, processed);
+
+    while (node) {
+        let cost = costs[node];
+        let children = graph[node];
+        for (let n in children) {
+            if (String(n) !== String(start)) {
+                let newCost = cost + children[n];
+                if (!costs[n] || costs[n] > newCost) {
+                    costs[n] = newCost;
+                    parents[n] = node;
+                }
+            }
+        }
+        processed.push(node);
+        node = lowestCostNode(costs, processed);
+    }
+
+    let optimalPath = [end];
+    let parent = parents[end];
+    while (parent) {
+        optimalPath.push(parent);
+        parent = parents[parent];
+    }
+    optimalPath.reverse();
+
+    const results = {
+        distance: costs[end],
+        path: optimalPath
+    };
+
+    return results;
+};
+
+// Looks for cost of paths & goes with cheapest in cost
+const lowestCostNode = (costs, processed) => {
+    return Object.keys(costs).reduce((lowest, node) => {
+        if (lowest === null || costs[node] < costs[lowest]) {
+            if (!processed.includes(node)) {
+                lowest = node;
+            }
+        }
+        return lowest;
+    }, null);
+};
+
+
+/**
+ * Get the floor number of a specific classroom. 
+ * Ex: H829 returns 8
+ * @param {*} name | classroom name
+ */
+export const getFloorNumber = (name) => {
+    if (name) {
+        const num = name.match(/\d+/g).map(Number);
+
+        if(num.length > 1) {
+            return num[0].toString();
+        }
+        if (num.toString().length == 3) {
+            return num.toString().charAt(0)
+        }
+        if (num.toString().length == 4) {
+            return num.toString().charAt(0) + num.toString().charAt(1);
+        }   
+    }
+}
+
+/**
+ * Only used to fetch information from data
+ * 
+ * Change the classroom name to H8XX because our
+ * data file stores uses associative arrays with H8XX as the key names. 
+ * Ex: H529 returns H829
+ * @param {*} name | classroom name
+ */
+export const changeClassName = (name) => {
```

Comment:
1. I would have called this method _ClassroomNameSanitizer_.  
2. What does this line does?  `const num = name.match(/\d+/g).map(Number);`

Reply:
It does not _sanitize_ the class name, it changes it to H8XX so we can fetch the data

Reply:
_ConvertToHall8Floor_ then. I feel that _changeClassName_ is too generic while you're doing something very specific. As a coder, I had to go in the definition of that method while reading this file because I didn't know what was going on.

Reply:
Will change!